docs/rst/use_cases/keys/keys.rst

```diff
@@ -0,0 +1,171 @@
+    Multiple service clients can use the same service,
+    but there can only be one service server for a given service.
+    Since each Turtle Node must have a service server to control goals,
+    it does not make sense to have multiple clients (e.g., controllers) using the same service server with different keys.
+
+
+Prerequisites
+-------------
+
+Familiarity with the following topics is recommended:
+
+* `Content Filter Topic <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/contentFilteredTopic.html>`__
+* :ref:`topic_keys`
+
+It is required to have previously installed Vulcanexus using one of the following installation methods:
+
+* :ref:`linux_binary_installation`
+* :ref:`linux_source_installation`
+* :ref:`docker_installation`
+
+
+Keyed Messages
+--------------
+
+The keyed Topics use the following message types to represent the turtles' velocity and pose:
+
+.. raw:: html
+
+    <div style="display: flex;">
+        <div style="flex: 1; padding: 10px;">
+            <p><b>docs_turtlesim::msg::KeyedPose</b></p>
+            <p>Represents the pose of a turtle with a key.</p>
+            <ul>
+                <li>key: Identifier of the turtle.</li>
+                <li>x: X-coordinate of the turtle's position.</li>
+                <li>y: Y-coordinate of the turtle's position.</li>
+                <li>theta: Orientation angle of the turtle.</li>
+            </ul>
+        </div>
+        <div style="flex: 1; padding: 10px;">
+            <img src="../../../_static/resources/use_cases/keys/keyed_pose.png" alt="KeyedPose" style="width: 220px;">
+        </div>
+    </div>
+
+.. raw:: html
+
+    <div style="display: flex;">
+        <div style="flex: 1; padding: 10px;">
+            <p><b>docs_turtlesim::msg::KeyedVelocity</b></p>
+            <p>Represents the velocity of a turtle with a key.</p>
+            <ul>
+                <li>key: Identifier of the turtle.</li>
+                <li>linear: Linear velocity of the turtle.</li>
+                <li>angular: Angular velocity of the turtle.</li>
+            </ul>
+        </div>
+        <div style="flex: 1; padding: 10px;">
+            <img src="../../../_static/resources/use_cases/keys/keyed_twist.png" alt="KeyedTwist" style="width: 320px;">
+        </div>
+    </div>
+
+
+Prepare the ROS 2 workspace
+---------------------------
+
+The ROS 2 package used in this use case is:
+
+* `docs_turtlesim <https://github.com/eProsima/vulcanexus/tree/main/code/turtlesim>`__: a simple ROS 2 package for teaching ROS 2 concepts.
+
+To create a new ROS 2 workspace and clone the `docs_turtlesim` package run:
+
+.. code-block:: bash
+
+    mkdir -p $HOME/ROS2-ws/src
+    cd $HOME/ROS2-ws/src
+    git clone --depth=1 https://github.com/eProsima/vulcanexus.git tmp_dir
+    cd tmp_dir
+    git archive --format zip --output ../turtlesim.zip HEAD:code/turtlesim
+    cd ..
+    unzip turtlesim.zip -d docs_turtlesim
+    rm -rf tmp_dir turtlesim.zip
```

Comment:
Just clone the repository (you can keep the `--depth=1`), copy the folder and then remove the repository. I do not see why you need to compress the folder to uncompress it afterwards